docs/dev/design_documents/gem_installation_via_metadata.md

```diff
@@ -0,0 +1,31 @@
+# Enable gem dependencies in cookbook metadata
+
+Support a 'gem' DSL method for cookbook metadata to create a dependency on a rubygem.  The
+gem will be installed via `chef_gem` after all the cookbooks are synchronized but before any
+other cookbook loading is done.
+
+## Motivation
+
+    As a Chef User,
+    I want to be able to use additional gems in libraries, attributes and resources,
+    to avoid complex workarounds and double-run converges.
+
+## Specification
+
+Allow users to specify additional gem dependencies like:
+
+```ruby
+gem "poise"
+gem "chef-sugar"
+gem "chef-provisioning"
+```
+
+In the `Chef::RunContext::CookbookCompiler#compile` method a phase will be added before `compile_libraries` which will install all of the gem declarations from all of the synchronized cookbooks before any other cookbook code is compiled.
+
+The implementation will use an in-memory bundler Gemfile which is constructed against all gem statements in all cookbooks which are in the `run_list`, solved
```

Comment:
There are two dependent clauses in a row that begin with "which". It seems like something is missing from this description, which would clarify the sentence.

Reply:
Both are technically correct statements. Do you have a suggestion for how we could better word this?

Reply:
```suggestion
The implementation will use an in-memory bundler Gemfile that is constructed against all gem statements in all cookbooks, which are in the `run_list`, solved
```

Reply:
I'll open a new PR for this, since this PR has been merged.